From a6a37b4845c94ddc7943f8f8eb8b9500aa4db4af Mon Sep 17 00:00:00 2001
From: hyphop <art@khadas.com>
Date: Mon, 2 Dec 2019 10:44:32 +0900
Subject: [PATCH] dts no need ethernet reset defination for khadas VIM2!

because broken `ifconfig eth0 down && ifconfig eth0 up` - iface never up again
> ifconfig: ioctl 0x8914 failed: Device or resource busy

$ dmesg -c
[  562.109408] meson8b-dwmac c9410000.ethernet eth0: PHY [0.2009087f:00] driver [RTL8211F Gigabit Ethernet]
[  562.109498] meson8b-dwmac c9410000.ethernet eth0: phy: setting supported 00,00000000,000062ff advertising 00,00000000,000062ff
[  562.224926] meson8b-dwmac c9410000.ethernet: Failed to reset the dma
[  562.225000] meson8b-dwmac c9410000.ethernet eth0: stmmac_hw_setup: DMA engine initialization failed
[  562.225053] meson8b-dwmac c9410000.ethernet eth0: stmmac_open: Hw setup failed

more details: https://forum.khadas.com/t/vim2-ethernet-problem-with-mainline-kernel/5817/7

NOTE: ethernet works again without reset defination
---
 arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts b/arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts
index f25ddd18a..76a41d182 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts
@@ -252,8 +252,7 @@
 
 		reset-assert-us = <10000>;
 		reset-deassert-us = <30000>;
-		reset-gpios = <&gpio GPIOZ_14 GPIO_ACTIVE_LOW>;
-
+		max-speed = <1000>;
 		interrupt-parent = <&gpio_intc>;
 		/* MAC_INTR on GPIOZ_15 */
 		interrupts = <25 IRQ_TYPE_LEVEL_LOW>;
-- 
2.17.1

